src/hdlc.rs

```diff
@@ -0,0 +1,97 @@
+use alloc::borrow::Cow;
+use hdlcparse::type3::HdlcFrame;
+use nom::{number::complete::u8, sequence::tuple};
+
+use crate::{DlmsDataLinkLayer, Error};
+
+enum Destination {
+  Unicast,
+  Broadcast,
+}
+
+enum MessageType {
+  Command,
+  Response,
+}
+
+struct LlcHeader {
+  destination: Destination,
+  message_type: MessageType,
+  quality: u8,
+}
+
+fn parse_llc_header(input: &[u8]) -> Option<(&[u8], LlcHeader)> {
```

Comment:
```suggestion
fn parse_llc_header(input: &[u8]) -> Result<(&[u8], LlcHeader), Error> {
```

I think returning a result here is a bit more readable, given the usage below currently adds an error with `ok_or` anyways.